nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/credentials/factory/CredentialPropertyDescriptors.java

```diff
@@ -89,4 +89,57 @@ private CredentialPropertyDescriptors() {}
             .sensitive(true)
             .build();
 
+    /**
+     * Specifies use of Service Account Impersonation Credentials
+     *
+     * @see <a href="https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials">
+     *     Google Service Account Impersonation
+     *     </a>
+     */
+    public static final PropertyDescriptor IMPERSONATE_SERVICE_ACCOUNT = new PropertyDescriptor.Builder()
+            .name("impersonate-service-account")
+            .displayName("Impersonate Service Account")
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .required(false)
+            .sensitive(false)
+            .description("The service account to impersonate.")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor IMPERSONATE_DELEGATION_CHAIN = new PropertyDescriptor.Builder()
+            .name("impersonate-delegation-chain")
+            .displayName("Impersonate Delegation Chain")
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .required(false)
+            .sensitive(false)
+            .description("A comma-separated list of chained delegates that required to grant the final access_token. "
+                    + "This property is valid only when using the service account impersonation strategy; otherwise, it is ignored.")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .dependsOn(IMPERSONATE_SERVICE_ACCOUNT)
+            .build();
+
+    public static final PropertyDescriptor IMPERSONATE_OAUTH_SCOPE = new PropertyDescriptor.Builder()
+            .name("impersonate-oauth-scope")
+            .displayName("Impersonate OAuth Scope")
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .required(true)
+            .sensitive(false)
+            .description("A comma-separated list of scopes to request during the authorization grant. "
+                    + "This property is valid only when using the service account impersonation strategy; otherwise, it is ignored.")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .dependsOn(IMPERSONATE_SERVICE_ACCOUNT)
+            .build();
+
+    public static final PropertyDescriptor IMPERSONATE_CREDENTIAL_LIFETIME = new PropertyDescriptor.Builder()
```

Comment:
What will happen when the processor is running for a period over the value defined in this property? Won't the access be revoked?

Reply:
Google's SDK will auto-refreshed by itself. if you want to revoke, need to set in IAM